backend/src/paymaster/index.ts

```diff
@@ -667,4 +669,102 @@ export class Paymaster {
+    isEp06: boolean,
+    chainId: number,
+    log?: FastifyBaseLogger
+  ) {
+    try {
+      const provider = new providers.JsonRpcProvider(bundlerRpcUrl);
+      const signer = new Wallet(privateKey, provider);
+      
+      let contract;
+      if(isEp06) {
+        contract = new ethers.ContractFactory(verifyingPaymasterAbi, verifyingPaymasterByteCode, signer);
+      } else {
+        contract = new ethers.ContractFactory(verifyingPaymasterV2Abi, verifyingPaymasterV2ByteCode, signer);
+      }
+
+      const etherscanFeeData = await getEtherscanFee(chainId);
+      let feeData;
+      if (etherscanFeeData) {
+        feeData = etherscanFeeData;
+      } else {
+        feeData = await provider.getFeeData();
+        feeData.gasPrice = feeData.gasPrice ? feeData.gasPrice.add(this.feeMarkUp) : null;
+        feeData.maxFeePerGas = feeData.maxFeePerGas ? feeData.maxFeePerGas.add(this.feeMarkUp) : null;
+        feeData.maxPriorityFeePerGas = feeData.maxPriorityFeePerGas ? feeData.maxPriorityFeePerGas.add(this.feeMarkUp) : null;
+      }
+
+      let tx;
+      if (!feeData.maxFeePerGas) {
+        tx = await contract.deploy(epAddr, signer.address, {gasPrice: feeData.gasPrice});
+      } else {
+        tx = await contract.deploy(
+          epAddr,
+          signer.address,
+          {
+            maxFeePerGas: feeData.maxFeePerGas ?? undefined,
+            maxPriorityFeePerGas: feeData.maxPriorityFeePerGas ?? undefined,
+            type: 2
+          }
+        );
+      }
+      await tx.deployed();
+      return { address: tx.address, hash: tx.deployTransaction.hash };
+    } catch (error) {
+      log?.error(`error while deploying verifying paymaster ${error}`);
+      throw new Error(ErrorMessage.FAILED_TO_DEPLOY_VP);
+    }
+  }
+
+  async addStake(
+    privateKey: string,
+    bundlerRpcUrl: string,
+    amount: string,
+    paymasterAddress: string,
+    chainId: number,
+    log?: FastifyBaseLogger
+  ) {
+    try {
+      const provider = new providers.JsonRpcProvider(bundlerRpcUrl);
+      const signer = new Wallet(privateKey, provider);
+
+      const contract = new ethers.Contract(paymasterAddress, verifyingPaymasterAbi, signer);
+
+      const etherscanFeeData = await getEtherscanFee(chainId);
+      let feeData;
+      if (etherscanFeeData) {
+        feeData = etherscanFeeData;
+      } else {
+        feeData = await provider.getFeeData();
+        feeData.gasPrice = feeData.gasPrice ? feeData.gasPrice.add(this.feeMarkUp) : null;
+        feeData.maxFeePerGas = feeData.maxFeePerGas ? feeData.maxFeePerGas.add(this.feeMarkUp) : null;
+        feeData.maxPriorityFeePerGas = feeData.maxPriorityFeePerGas ? feeData.maxPriorityFeePerGas.add(this.feeMarkUp) : null;
+      }
+
+      let tx;
+      if (!feeData.maxFeePerGas) {
+        tx = await contract.addStake("10", {value: ethers.utils.parseEther(amount), gasPrice: feeData.gasPrice});
+      } else {
+        tx = await contract.addStake(
+          "10",
```

Comment:
i have hardcoded the unstakeDelaySec param to be sent to smart contract as 10 @ch4r10t33r 
let me know if this is something to be accepted from client side.